modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/FilterToElastic.java

```diff
@@ -1217,11 +1220,26 @@ private void visitLiteralGeometry(Literal expression) throws IOException {
             final CoordinateSequence coordinates = linearRing.getCoordinateSequence();
             currentGeometry = factory.createLineString(coordinates);
         }
-
-        final String geoJson = GeoJSONWriter.toGeoJSON(currentGeometry);
+        int maxDecimals = getMaxDecimalsForGeometry(currentGeometry);
+        final String geoJson = GeoJSONWriter.toGeoJSON(currentGeometry, maxDecimals);
         currentShapeBuilder = mapReader.readValue(geoJson);
     }
 
+    private int getMaxDecimalsForGeometry(Geometry currentGeometry) {
+        Envelope envelope = currentGeometry.getEnvelopeInternal();
+        double min = Math.min(Math.abs(envelope.getWidth()), Math.abs(envelope.getHeight()));
+        if (min == 0) {
+            LOGGER.log(
+                    Level.WARNING,
+                    "BBox Geometry has no width or height, it is either a point or a line.");
+            return 0;
+        } else if (min >= 0.000001) {
+            return JtsModule.DEFAULT_MAX_DECIMALS;
```

Comment:
Did you try zooming in a lot and doing a GetFeatureInfo? Eventually using only 6 decimals will not be enough and the request will fail. The idea was to adjust the decimals so that one would not get an error, at least until we don't reach the full precision at which we can represent doubles (16 decimals)